include/envoy/server/listener_manager.h

```diff
@@ -52,10 +52,9 @@ class ListenerComponentFactory {
    * @param bind_to_port supplies whether to actually bind the socket.
    * @return Network::SocketSharedPtr an initialized and potentially bound socket.
    */
-  virtual Network::SocketSharedPtr
-  createListenSocket(Network::Address::InstanceConstSharedPtr address,
-                     Network::Address::SocketType socket_type,
-                     const Network::Socket::OptionsSharedPtr& options, bool bind_to_port) PURE;
+  virtual Network::SocketSharedPtr createListenSocket(
+      Network::Address::InstanceConstSharedPtr address, Network::Address::SocketType socket_type,
+      const Network::Socket::OptionsSharedPtr& options, bool bind_to_port, bool reuse_port) PURE;
```

Comment:
Please update the doc above for the new field. Also, I have a strong preference that we replace the 2 bool params with a param struct. This will avoid having to update a ton of different call sites and tests next time we add a param. Since you already have to do it for this change it seems less painful for you to do it now. :)